sdk/spring/azure-spring-integration-servicebus/src/main/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverter.java

```diff
@@ -73,32 +75,48 @@ protected ServiceBusMessage fromByte(byte[] payload) {
     @Override
     protected void setCustomHeaders(MessageHeaders headers, ServiceBusMessage message) {
 
+        Set<String> copyHeaders = new HashSet<String>();
+        headers.forEach((key, value) -> {
+            copyHeaders.add(key);
+        });
```

Comment:
headers.keySet()

Reply:
The keySet method of org.springframework.messaging.MessageHeaders returns an unmodified Set. I have updated the copyHeaders as a map and use the putAll API.